linker/Linker.Steps/SweepStep.cs

```diff
@@ -208,7 +208,8 @@ void SweepReferences (AssemblyDefinition assembly, AssemblyDefinition target)
 					}
 					break;
 				}
-				return;
+				// earlier removal from `references` requires an adjustment
+				i--;
```

Comment:
I think it'd make it more clear if this was done as part of `references.RemoveAt (i);`